src/components/clusterWizard/ClusterDetails.tsx

```diff
@@ -0,0 +1,273 @@
+          ? validJSONSchema
+          : validJSONSchema.required('Pull secret must be provided.'),
+        baseDnsDomain: dnsNameValidationSchema(initialValues.baseDnsDomain),
+      }),
+    [],
+  );
+
+  const handleSubmit = async (
+    values: ClusterDetailsValues,
+    formikActions: FormikHelpers<ClusterDetailsValues>,
+  ) => {
+    clearAlerts();
+
+    // async validation for cluster name - run only on submit
+    // TODO(jtomasek): Update this to validate combination of cluster name + dns domain
+    try {
+      const { data: clusters } = await getClusters();
+      const names = clusters.map((c) => c.name).filter((n) => n !== cluster.name);
+      if (names.includes(values.name)) {
+        return formikActions.setFieldError('name', `Name "${values.name}" is already taken.`);
+      }
+    } catch (e) {
+      captureException(e, 'Failed to perform unique cluster name validation.');
+    }
+
+    try {
+      const params = _.omit(values, ['useRedHatDnsService']);
+
+      if (pullSecretSet) {
+        delete params.pullSecret;
+      }
+
+      const { data } = await patchCluster(cluster.id, params);
+      dispatch(updateCluster(data));
+      setCurrentStepId('cluster-configuration');
+    } catch (e) {
+      handleApiError<ClusterUpdateParams>(e, () =>
+        addAlert({ title: 'Failed to update the cluster', message: getErrorMessage(e) }),
+      );
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({ submitForm, isSubmitting, isValid, values, setFieldValue }) => {
+        const { name: clusterName, baseDnsDomain, useRedHatDnsService } = values;
+
+        const baseDnsHelperText = (
+          <>
+            All DNS records must be subdomains of this base and include the cluster name. This
+            cannot be changed after cluster installation. The full cluster address will be: <br />
+            <strong>
+              {clusterName || '[Cluster Name]'}.{baseDnsDomain || '[example.com]'}
+            </strong>
+          </>
+        );
+
+        const toggleRedHatDnsService = (checked: boolean) =>
+          setFieldValue('baseDnsDomain', checked ? managedDomains.map((d) => d.domain)[0] : '');
+
+        const form = (
+          <Grid hasGutter>
+            <GridItem span={12} lg={10} xl={9} xl2={7}>
+              <Form
+                id="wizard-cluster-details__form"
+                onKeyDown={(e) => {
+                  if (e.key === 'Enter') {
+                    submitForm();
+                  }
+                }}
+              >
+                <InputField ref={nameInputRef} label="Cluster Name" name="name" isRequired />
+                <SelectField
+                  label="OpenShift Version"
+                  name="openshiftVersion"
+                  options={[{ label: openshiftVersion, value: openshiftVersion }]}
+                  // getHelperText={getOpenshiftVersionHelperText}
```

Comment:
Why is this commented out?

Reply:
At this point the field is disabled, I am going to update this in a follow up when I mix this with NewClusterPage functionality. Also I think I'll need to fetch the versions here and try to compare to what is set on the cluster.